src/tasks/DockerTaskEndEventListener.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface DockerTaskEvent {
+    name: string,
+    success: boolean,
+    error?: string
+}
+
+type DockerTaskEventListener = { (taskEvent: DockerTaskEvent): void; };
+
+export class DockerTaskEndEventListener {
```

Comment:
VSCode has an `EventEmitter<T>` class that we be using or deriving from.

Reply:
That's definitely handy, I think we can just use it there's no need to extend it.

Reply:
Yeah. I honestly don't really know how it works, but it seems like it should do what we need. :smile:

Reply:
Additionally: it produces `Disposable`s, which we might need to manually dispose of, or at least attach to `context.subscriptions`.

Reply:
Yeah, it's perfect to what we need here. So I switched to using that, with appropriately disposing the listener after we're done with it.